### PR TITLE
Catch socket exceptions

### PR DESCRIPTION
**Pull request by foman**
## Description
When to access the endpoint , it may throw exceptions, according to [msdn](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.remoteendpoint?view=netframework-4.7.2#remarks). 
It indeed happened in my windows 10 machine, so it is better to catch it.

## Related issue
##1607 Failed to negotiate protocol. Wait for response timeout

### CODE REVIEW

File: src/Microsoft.TestPlatform.CommunicationUtilities/TcpClientExtensions.cs [hunk 1]
```diff
@@ -25,8 +25,20 @@ internal static class TcpClientExtensions
...
+            var localEndPoint = string.Empty;
+            try
+            {
+                remoteEndPoint = client.Client.RemoteEndPoint.ToString();
+                localEndPoint = client.Client.LocalEndPoint.ToString();
+            }
+            catch (SocketException socketException)
+            {
+                EqtTrace.Error(
+                        "TcpClientExtensions.MessageLoopAsync: failed to access endpoint of the socket {0}",
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CommunicationUtilities/TcpClientExtensions.cs]: "Failed to access the endpoint due to socket error : {0}"

Comment (singhsarab) [reply src/Microsoft.TestPlatform.CommunicationUtilities/TcpClientExtensions.cs]: Why was the null check removed ?

Comment (foman) [reply src/Microsoft.TestPlatform.CommunicationUtilities/TcpClientExtensions.cs]: OK, fixed

Comment (singhsarab) [reply src/Microsoft.TestPlatform.CommunicationUtilities/TcpClientExtensions.cs]: @foman Apologies for the misunderstanding.
We should keep the method signature. I expect the following.

TcpClientExtensions.MessageLoopAsync : Failed to access the endpoint due to socket error : {0}

Comment (foman) [reply src/Microsoft.TestPlatform.CommunicationUtilities/TcpClientExtensions.cs]: Ok

**Review by singhsarab — APPROVED** [review]

**Comment by foman** [comment]
No idea why the UT failed.

**Comment by singhsarab** [comment]
@dotnet-bot test this please

**Pull request merged**